src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1036,6 +1037,39 @@ class spell_warr_t3_prot_8p_bonus : public AuraScript
     }
 };
 
+// 389603 - Unbridled Ferocity
+class spell_warr_unbridled_ferocity : public AuraScript
+{
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_RECKLESSNESS })
+            && ValidateSpellEffect({ { spellInfo->Id, EFFECT_1 } })
+            && spellInfo->GetEffect(EFFECT_0).IsAura(SPELL_AURA_DUMMY)
+            && spellInfo->GetEffect(EFFECT_1).IsAura(SPELL_AURA_DUMMY);
+    }
+
+    void HandleProc(ProcEventInfo& /*eventInfo*/) const
+    {
+        int32 durationMs = GetAura()->GetEffect(EFFECT_1)->GetAmount();
```

Comment:
you can also use Milliseconds type and then .count()

Reply:
that'd be unnecessary casting